docs/manual-configuration.md

```diff
@@ -0,0 +1,135 @@
+---
+id: manual-configuration
+title: Manual configuration
+---
+
+The `lsif-java index` command does a best-effort to automatically index a given
+codebase. The automatic step may not always work. The purpose of this page is to
+document the steps to manually configure lsif-java when automatic indexing
+fails.
+
+## Overview
+
+Indexing a codebase consists of two independent steps:
+
+- Compile the codebase with the SemanticDB compiler plugin.
+- Generate LSIF index from SemanticDB files.
+
+![A three stage pipeline that starts with a list of Java sources, creates a list of SemanticDB files that then become a single LSIF index.](docs/img/semanticdb-javac-pipeline.svg)
+
+## Step 1: Add SemanticDB compiler plugin to the classpath
```

Comment:
Should we have consistent first letter capitalization or is this intentionally capitalized?